feature/experimental/bgp/ate_tests/bgp_remove_private_as/bgp_remove_private_as_test.go

```diff
@@ -0,0 +1,384 @@
+	port1 := ate.Port(t, "port1")
+	topo := ate.Topology().New()
+	iDut1 := topo.AddInterface(ateSrc.Name).WithPort(port1)
+	iDut1.IPv4().WithAddress(ateSrc.IPv4CIDR()).WithDefaultGateway(dutSrc.IPv4)
+
+	port2 := ate.Port(t, "port2")
+	iDut2 := topo.AddInterface(ateDst.Name).WithPort(port2)
+	iDut2.IPv4().WithAddress(ateDst.IPv4CIDR()).WithDefaultGateway(dutDst.IPv4)
+
+	// Setup ATE BGP route v4 advertisement.
+	bgpDut1 := iDut1.BGP()
+	bgpDut1.AddPeer().WithPeerAddress(dutSrc.IPv4).WithLocalASN(ateAS1).
+		WithTypeExternal()
+
+	bgpDut2 := iDut2.BGP()
+	bgpDut2.AddPeer().WithPeerAddress(dutDst.IPv4).WithLocalASN(ateAS2).
+		WithTypeExternal()
+
+	bgpNeti1 := iDut1.AddNetwork("bgpNeti1")
+	bgpNeti1.IPv4().WithAddress(advertisedRoutesv4CIDR).WithCount(routeCount)
+	bgpNeti1.BGP().WithNextHopAddress(ateSrc.IPv4)
+
+	if asSEQMode {
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSEQ()
+	} else {
+		// TODO : SET mode is not working
+		// https://github.com/openconfig/featureprofiles/issues/1659
+		bgpNeti1.BGP().AddASPathSegment(asSeg...).WithTypeSET()
+	}
+
+	t.Logf("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+
+	return topo
+}
+
+type bgpNeighbor struct {
+	as         uint32
+	neighborip string
+	isV4       bool
+	peerGrp    string
+}
+
+// verifyBGPAsPath is to Validate AS Path attribute using bgp rib telemetry on ATE.
+func verifyBGPAsPath(t *testing.T, dut *ondatra.DUTDevice, ate *ondatra.ATEDevice, asSeg []uint32, removeASPath bool) {
+	at := gnmi.OC()
+	rib := at.NetworkInstance(ateDst.Name).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "0").Bgp().Rib()
+	prefixPath := rib.AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Ipv4Unicast().
+		NeighborAny().AdjRibInPre().RouteAny().WithPathId(0).Prefix()
+
+	gnmi.WatchAll(t, ate, prefixPath.State(), time.Minute, func(v *ygnmi.Value[string]) bool {
+		_, present := v.Val()
+		return present
+	}).Await(t)
+
+	var wantASSeg = []uint32{dutAS, ateAS1}
+
+	if removeASPath {
+		for _, as := range asSeg {
+			if as < 64512 {
+				wantASSeg = append(wantASSeg, as)
+			}
+		}
+	} else {
+		wantASSeg = append(wantASSeg, asSeg...)
+	}
+
+	gotASSeg, ok := gnmi.WatchAll(t, ate, rib.AttrSetAny().AsSegmentAny().State(), 1*time.Minute, func(v *ygnmi.Value[*oc.NetworkInstance_Protocol_Bgp_Rib_AttrSet_AsSegment]) bool {
+		val, present := v.Val()
+		return present && cmp.Diff(val.Member, wantASSeg) == ""
+	}).Await(t)
+	if !ok {
+		t.Errorf("Obtained AS path on ATE is not as expected, gotASSeg %v, wantASSeg %v", gotASSeg, wantASSeg)
+	}
+}
+
+// configreRoutePolicy adds route-policy config
+func configureRoutePolicy(t *testing.T, dut *ondatra.DUTDevice, name string, pr oc.E_RoutingPolicy_PolicyResultType) {
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pd := rp.GetOrCreatePolicyDefinition(name)
+	st := pd.GetOrCreateStatement("id-1")
+	stc := st.GetOrCreateConditions()
```

Comment:
Hi @cprabha, can we remove lines `302, 303` as these are not needed in any other BGP tests.

Reply:
With BGP policy to accept routes without any match condition, all the routes including direct will be advertised. As this testcase has a check for sent prefixes count , it can be verified with exact value. Since the expectation is to verify BGP routes, have added match condition based on BGP routes. 
In other BGP tests there is no check for sent prefixes or  it is checking for "zero" sent prefixes (no route advertisement). Hence this is not required in other tests.

Reply:
By default only the eBGP learnt routes learnt from the AS100/Port 1 will be advertised to AS200/Port2 ?
For us unless explicitly specified by by config , the direct/connected routes are not advertised in the BGP.

Reply:
No , we support RFC8212 thats why require a RPL to be configured under the eBGP Nbr/Peer-group.
My comment was regards to the condition below. We don’t require this, since only the eBGP learnt routes will be advertised .Direct/connected routes are not advertised in BGP unless explicitly specified in the config.

stc := st.GetOrCreateConditions()
stc.InstallProtocolEq = oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP

Reply:
ok, understood. Lets merge this as is for now. I have started an internal conversation to get an agreement on how to get a longterm fix. @vosipchu will suggest next steps as soon as we get an agreement to guide on future pull.

@dplore for viz.

Reply:
Sure thanks. 
Also, if the idea is to filter on BGP only routes in the RPL then it can be made conditional based on a prefix-set as well.